python/tchannel/tornado/connection.py

```diff
@@ -518,6 +528,11 @@ def _stream(self, context, message_factory):
 
     @tornado.gen.coroutine
     def post_response(self, response):
+
+        # event: send_response
+        if self.tchannel:
+            self.tchannel.event_emitter.fire(EventType.send_response, response)
```

Comment:
I thought about this before. I didn't do this way because it requires the hook to have each event methods. In side the require methods, they basically have the same logic. 

I take the pattern that you write in the BaseRequestHandler method, doing "handle_"+name style.  I think this pattern is quite concise and reuse the code.

I agree both will work.